chapters/slang.adoc

```diff
@@ -0,0 +1,744 @@
+
+Use Slang's metaprogramming capabilities for more powerful shader generation:
+
+[source,slang]
+----
+// Define shader permutations using compile-time parameters
+[shader("vertex")]
+[CompileTimeConstant(name="USE_NORMAL_MAPPING", type="bool")]
+[CompileTimeConstant(name="LIGHT_COUNT", type="int")]
+VSOutput vertexShader(VSInput input) {
+    VSOutput output;
+    // Base implementation
+    #if USE_NORMAL_MAPPING
+    // Normal mapping specific code
+    #endif
+    for (int i = 0; i < LIGHT_COUNT; i++) {
+        // Per-light calculations
+    }
+    return output;
+}
+----
+
+=== Common Migration Challenges
+
+==== Resource Binding Compatibility
+
+**Challenge**: Slang's resource binding model differs from HLSL's register-based approach.
+
+**Solution**:
+
+- Use Slang's `register` compatibility syntax during transition
+- Gradually migrate to parameter blocks
+- Update shader binding code in your application
+
+==== Module Organization
+
+**Challenge**: Deciding how to organize code into modules.
+
+**Solution**:
+- Group related functionality into modules
+- Use hierarchical naming (e.g., `Rendering.Lighting`)
+- Start with coarse-grained modules and refine as needed
+
+==== Interface Performance
+
+**Challenge**: Concerns about runtime performance of interfaces.
+
+**Solution**:
+- Interfaces are often resolved at compile-time
+- Use interfaces for flexibility in high-level code
+- Profile performance-critical paths
+
+==== Compilation Pipeline Integration
+
+**Challenge**: Integrating Slang into existing build systems.
+
+**Solution**:
+- Create wrapper scripts to maintain command-line compatibility
+- Update build tools to support both HLSL and Slang during transition
+- Consider using Slang's API for deeper integration
+
+=== Best Practices for Migration
+
+1. **Incremental Approach**: Migrate one shader or shader module at a time
+2. **Maintain Compatibility**: Use Slang's HLSL compatibility features during transition
+3. **Test Thoroughly**: Verify visual output after each migration step
+4. **Refactor Gradually**: Start with simple syntax changes, then introduce advanced features
+5. **Leverage Modules**: Use the module system to improve code organization
+6. **Document Changes**: Keep track of migration decisions and patterns
+7. **Performance Profiling**: Monitor performance before and after migration
+
+== Migration Guide: from GLSL to Slang
+
+Migrating from GLSL to Slang involves more significant changes than migrating from HLSL, as the languages have different syntax and programming models. However, Slang's powerful features can make the transition worthwhile for many projects.
+
+[TIP]
+====
+For a comprehensive guide on migrating from GLSL to Slang, see the official link:https://docs.shader-slang.org/en/latest/coming-from-glsl.html[Coming from GLSL] documentation.
+====
+
+=== Key Differences to Consider
+
+* *Syntax*: GLSL uses a more procedural style similar to C, while Slang uses an object-oriented style similar to C++
```

Comment:
Hmm... Not sure if this is a good first impression to give, considering Slang has deprecated some OOP features like struct inheritance.